packages/admin/cms-admin/src/generator/future/generateForm/generateFormField.ts

```diff
@@ -41,6 +41,15 @@ export function generateFormField({
     if (!introspectionField) throw new Error(`didn't find field ${name} in gql introspection type ${gqlType}`);
     const introspectionFieldType = introspectionField.type.kind === "NON_NULL" ? introspectionField.type.ofType : introspectionField.type;
 
+    const fieldIsOptionalInApi = isFieldOptionalInApi({ name: String(config.name), gqlIntrospection, gqlType });
+    const optionalRender = config.optionalRenderProp && fieldIsOptionalInApi;
+    if (config.optionalRenderProp && !fieldIsOptionalInApi) {
+        console.warn(
+            `Field ${String(
+                config.name,
+            )}: Required input can not be optionalRender. Try generating a second form without this field to enable providing a value via prop.`,
+        );
```

Comment:
I thinks that's a good idea.

but the main problem with throwing errors in admin-generator is currently that it generates all files at once. so a throw error for a single config effects all configs. it's a bit better since we only delete the current file before we generate, but it still wont update the rest.

so if we decide to throw errors we should also do something to only fail for this single configuration.

Reply:
I think this should be refactored in a different pull-request.